state/event_table.go

```diff
@@ -259,6 +259,25 @@ func (t *EventTable) SelectStrippedEventsByIDs(txn *sqlx.Tx, verifyAll bool, ids
 
 }
 
+// SelectUnknownEventIDs accepts a list of event IDs and returns the subset of those which are not known to the DB.
+// The order of event IDs in the return value is not guaranteed.
+func (t *EventTable) SelectUnknownEventIDs(txn *sqlx.Tx, maybeUnknownEventIDs []string) ([]string, error) {
+	queryStr := `
+	WITH maybe_unknown_events(event_id) AS (SELECT unnest($1::text[]))
+	SELECT event_id
+	FROM maybe_unknown_events LEFT JOIN syncv3_events USING(event_id)
+	WHERE event_nid IS NULL;`
```

Comment:
I did test this in a postgres playground, but I think this might be worth a unittest.

Reply:
We don't use this style of query anywhere in the proxy atm. I'm guessing you do this so you can "select then negate" all on the DB side? I'd expected to see something more like:
```
SELECT event_id FROM syncv3_events WHERE room_id=$1 AND event_id=ANY($2)
```
but then you'd need to negate on the Go side to find the missing ones. I wonder which is better.

Reply:
Yeah, the thinking was basically to get the DB to tell me which events are unknown, rather than getting it to tell me which events are unknown and then computing the complement.

I don't think it makes a huge difference if the state block is small. But we may hit this block with a poller initial sync, in which case we expect few events to be unknown. It'd be nice to get those few returned to us, rather than the full state back.